modules/install/pages/upgrade-procedure-selection.adoc

```diff
@@ -0,0 +1,178 @@
+If acceptable performance can indeed be maintained under such conditions, the appropriate number of nodes should be removed from the cluster, and a full rebalance performed.
+From this point, the diminished cluster continues to serve data at lower but acceptable performance-levels; while the removed nodes are used as spares throughout the upgrade.
+This is described below, in xref:install:upgrade-strategies.adoc#cluster-online-swap-rebalance-with-capacity-reduced[Cluster Online: Swap Rebalance with Capacity Reduced].
+
+[#using-graceful-failover]
+==== Using Graceful Failover
+
+If it is possible to remove one or more Data Service nodes from the cluster, and run the cluster at reduced capacity for the duration of the upgrade procedure, xref:manage:manage-nodes/failover-graceful.adoc[Graceful Failover] can be used to bring individual Data Service nodes out of the cluster, and so allow them to be upgraded.
+The Graceful Failover procedure ensures that all the cluster's active vBuckets continue to be available, on the remaining Data Service nodes, after the node to be upgraded has been failed over.
+Subsequently, the upgraded node is restored to the cluster using xref:learn:clusters-and-availability/recovery.adoc#delta-recovery[Delta Recovery].
+
+The main advantages of Graceful Failover and Delta Recovery in this context are simpler management and a lower consumption of cluster-resources: since to upgrade a node, no _spare_ node is required, and no copying of data across nodes (such as is performed by rebalance) need occur.
+The main constraint of this option is that it can _only_ be used for nodes running the Data Service alone.
+
+Further considerations are provided in xref:learn:clusters-and-availability/graceful-failover.adoc#advantages-and-disadvantages[Advantages and Disadvantages].
+(Note also that this option is not available when choosing to upgrade with _net-new_ systems &#8212; as in many Cloud-based deployments; since the new Data Service nodes do not have previous nodes’ data in place.)
+
+[#summaries-of-upgrade-procedures]
+== Summaries of Upgrade Procedures
+
+In accordance with the dependency-definitions provided above, in xref:install:upgrade-strategies.adoc#upgrade-and-availability[Upgrade and Availability], this section summarizes the individual, recommended procedures for upgrading a Couchbase-Server cluster.
+All clusters are assumed to be _multi-node_ clusters, except in the final section, xref:install:upgrade-strategies.adoc#upgrade-for-single-node-clusters[Upgrade for Single-Node Clusters].
+
+[#online-upgrade]
+=== Upgrade with Cluster Online
+
+For a multi-node cluster, an _online upgrade_ means that the cluster continues to serve data while its nodes are progressively upgraded.
+Online upgrade can be performed in either of the following ways.
+
+[#cluster-online-swap-rebalance-with-capacity-reduced]
+==== Cluster Online: Swap Rebalance at Full Capacity
```

Comment:
I wonder if we should make a recommendation for cloud environments, given that is where most of our customers operate and with what marketing is positioning with.